asteroid/walk.py

```diff
@@ -628,14 +627,34 @@ def store_at_ix(structure_val, ix, value):
     else:
         raise ValueError("'{}' is not a mutable structure".format(structure_val[0]))
 
+    # Next, we do the actual memory storage operation
 
-    # index into memory and set the value
+    # If it's just an integer, index into that location and
+    # set the value
     if ix_val[0] == 'integer':
         memory[ix_val[1]] = value
         return
 
+    # otherwhise, if the index is a list
     elif ix_val[0] == 'list':
-        raise ValueError("slicing in patterns not supported")
+        # Get the l/rval
+        (LIST, lval) = ix_val
+        (LIST_r, rval) = value
```

Comment:
**Have you considered** that some value structures are not pairs (I seem to remember).  In that case statement
```
(LIST_r, rval) = value
```
will fail with a generic Python pattern match error which will stump users.  So, before you do the pattern matching it might be worthwhile to test `value[0]` explicitly whether or not it is a list...just move your test code up a few lines...

Reply:
Good point! I believe I've fixed this now: https://github.com/asteroid-lang/asteroid/pull/121/commits/8386db72afa2a8561227949a9cb00aa6aa848eeb